services/system/StagedTx/plugin/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+use bindings::host::common::{client as Client, server as Server, types::Error};
+use bindings::transact::plugin::{hooks::hook_tx_transform_label, intf::add_action_to_transaction};
+use psibase::fracpack::Pack;
+use psibase::services::staged_tx::action_structs::propose;
+use psibase::{AccountNumber, Checksum256, Hex, MethodNumber};
+use staged_tx::action_structs::*;
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+mod errors;
+use errors::ErrorType;
+
+fn validate_account(account: &str) -> Result<(), Error> {
+    match get_account(account) {
+        Ok(Some(_)) => Ok(()),
+        Ok(None) => Err(ErrorType::InvalidAccount(account.to_string()).into()),
+        Err(e) => Err(e),
+    }
+}
+
+fn get_assert_caller(context: &str, allowed_apps: &[&str]) -> Result<(), Error> {
+    let caller = Client::get_sender_app();
+    let caller_app = caller.app.as_ref().unwrap();
+    if caller.app.is_none() || !allowed_apps.contains(&caller_app.as_str()) {
+        return Err(ErrorType::Unauthorized(context.to_string(), caller_app.to_string()).into());
+    }
+    Ok(())
+}
+
+#[derive(Deserialize, Serialize)]
+struct StagedTxDetails {
+    data: StagedTxData,
+}
+
+#[derive(Deserialize, Serialize)]
+struct StagedTxData {
+    details: StagedTxDetailsInner,
+}
+
+#[derive(Deserialize, Serialize)]
+struct StagedTxDetailsInner {
+    txid: String,
+    #[serde(rename = "proposeBlock")]
+    propose_block: u32,
+    #[serde(rename = "proposeDate")]
+    propose_date: String,
+    proposer: String,
+    #[serde(rename = "actionList")]
+    action_list: ActionList,
+}
+
+#[derive(Deserialize, Serialize)]
+struct ActionList {
+    actions: Vec<ActionDetails>,
+}
+
+#[derive(Deserialize, Serialize)]
+struct ActionDetails {
+    sender: String,
+    service: String,
+    method: String,
+    #[serde(rename = "rawData")]
+    raw_data: String,
+}
+
+fn get_staged_txid(id: u32) -> Result<Checksum256, Error> {
+    let query = format!(
+        r#"query {{
+            details(id: {id}) {{
+                txid
+                proposeBlock
+                proposeDate
+                proposer
+                actionList {{
+                    actions {{
+                        sender
+                        service
+                        method
+                        rawData
+                    }}
+                }}
+            }}
+        }}"#,
+        id = id
+    );
+
+    let details = Server::post_graphql_get_json(&query).unwrap();
+    let details = serde_json::from_str::<StagedTxDetails>(&details).unwrap();
+    let txid = details.data.details.txid;
+    if txid.len() % 2 != 0 {
+        return Err(ErrorType::InvalidTxId.into());
+    }
+    let txid: Vec<u8> = (0..txid.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&txid[i..i + 2], 16).unwrap())
+        .collect();
+    let txid: [u8; 32] = txid.try_into().map_err(|_| ErrorType::InvalidTxId)?;
```

Comment:
Can't all of this be handled by using `Checksum256` instead of `String` for the field?

Reply:
When I make a sql query for the event with `WHERE txid='HEXDATA'` when the type is `Checksum256`, I always get back an empty array.

Reply:
I don't understand. Aren't the types here only used for deserializing? How do they affect the query?

Reply:
Also, try `WHERE txid=X'HEXDATA'` (a blob literal)

Reply:
This worked. Updated